synfig-studio/src/gui/render.cpp

```diff
@@ -339,39 +374,116 @@ RenderSettings::on_render_pressed()
 		catch(std::runtime_error& x)
 		{
 			canvas_interface_->get_ui_interface()->error(_("Unable to determine proper target from filename."));
-			return;
+			return false;
 		}
 	}
 
 	if(filename.empty() && calculated_target_name!="null")
 	{
 		canvas_interface_->get_ui_interface()->error(_("A filename is required for this target"));
-		return;
+		return false;
 	}
-
-	hide();
-
-	render_passes.clear();
-	if (toggle_extract_alpha.get_active())
+	
+	//Retrieve current render settings
+	RendDesc rend_desc(widget_rend_desc.get_rend_desc());
+	
+	//Check format which could have an image sequence as output
+	//If format is selected in comboboxtext_target
+	std::map<std::string,std::string> ext_multi = {{"bmp",".bmp"},{"cairo_png",".png"},
+					{"imagemagick",".png"}, {"jpeg",".jpg"},{"mng",".mng"},
+					{"openexr",".exr"},{"png",".png"},{"ppm",".ppm"}};
+	
+	std::map<std::string,std::string>::iterator ext_multi_it;
+	ext_multi_it = ext_multi.find(calculated_target_name);
+	
+	//calculated_target_name is a candidate with known output target (not Auto)
+	if(ext_multi_it != ext_multi.end())
```

Comment:
Do you see both if-else branches share a lot of same code?
Can you "optimize" it? Avoiding code repetition is good.